### ISSUE TITLE
Error when running with FastAPI

### ISSUE THREAD

**Issue by grizzly-monkey** [nonchan7720/sqs-polling#22]
```
ERROR:    Traceback (most recent call last):
  File "/opt/lk/eda-services/verifire_service/.venv/lib/python3.11/site-packages/starlette/routing.py", line 677, in lifespan
    async with self.lifespan_context(app) as maybe_state:
  File "/opt/lk/eda-services/verifire_service/.venv/lib/python3.11/site-packages/starlette/routing.py", line 566, in __aenter__
    await self._router.startup()
  File "/opt/lk/eda-services/verifire_service/.venv/lib/python3.11/site-packages/starlette/routing.py", line 654, in startup
    await handler()
  File "/opt/lk/eda-services/verifire_service/verifire_service/web/lifetime.py", line 35, in _startup
    main("verifire_service.services.s3_upload_poller.s3_upload_task")
  File "/opt/lk/eda-services/verifire_service/.venv/lib/python3.11/site-packages/sqs_polling/main.py", line 46, in main
    _handler(p)
  File "/opt/lk/eda-services/verifire_service/.venv/lib/python3.11/site-packages/sqs_polling/polling.py", line 123, in _handler
    loop.run_forever()
  File "uvloop/loop.pyx", line 1377, in uvloop.loop.Loop.run_forever
  File "uvloop/loop.pyx", line 518, in uvloop.loop.Loop._run
RuntimeError: this event loop is already running.

ERROR:    Application startup failed. Exiting.
```

code 

```
@polling(
    queue_name="service-s3",
    max_workers=5,
    max_number_of_messages=10,
)
def s3_upload_task(
    self: Polling,
    message_body: str,
    message_attribute: dict[str, Any] | None,
    message_group_id: str | None,
    message_duplication_id: str | None,
):
    import json
    from pprint import pprint

    data: dict[str, Any] = json.loads(message_body)
    data.update(message_attribute or {})
    pprint(data)
```

**Comment by nonchan7720** [comment]
Thanks for creating the Issue.
Can you tell us a little more about how it is used in combination with FastAPI?

**Comment by nonchan7720** [comment]
How about this?
https://github.com/nonchan7720/sqs-polling/pull/24/files#diff-8c7c9c5c527d60e8e48ce37d55cc11ed39f3b4d4cf678e239aeec67ad3bfad45

**Comment by nonchan7720** [comment]
@grizzly-monkey 


How about this?
https://github.com/nonchan7720/sqs-polling/pull/24/files#diff-8c7c9c5c527d60e8e48ce37d55cc11ed39f3b4d4cf678e239aeec67ad3bfad45

**Comment by grizzly-monkey** [comment]
> Thanks for creating the Issue. Can you tell us a little more about how it is used in combination with FastAPI?

Thanks for your fast response... I am trying to consume messages from SQS in our fast api application.
For this  i am using startup event of fast api to start consuming the SQS messages like this

```
    @app.on_event("startup")
    async def _startup() -> None:  # noqa: WPS430
        app.middleware_stack = None
        if not broker.is_worker_process:
            await broker.startup()
        app.middleware_stack = app.build_middleware_stack()

        db.connect(path=settings.db["uri"], pem=settings.db["secret_key"])
        main("verifire_service.services.s3_upload_poller.s3_upload_task") # here is the queue consumer start
        pass  # noqa: WPS420

    return _startup
```

**Comment by grizzly-monkey** [comment]
i am using boiler plate code from
https://github.com/s3rius/FastAPI-template
for creating our application if this helps

**Comment by grizzly-monkey** [comment]
I am not sure of right fix but i can tell if i comment out 
form main.py main()
```
    # loop.run_forever()
    # loop.close()
```

it works as needed , may be becuase the fast-api template i am using already is running an event loop of uvicorn uvloop ? 🤔 



```
from typing import Any

from gunicorn.app.base import BaseApplication
from gunicorn.util import import_app
from uvicorn.workers import UvicornWorker as BaseUvicornWorker

try:
    import uvloop  # noqa: WPS433 (Found nested import)
except ImportError:
    uvloop = None  # type: ignore  # noqa: WPS440 (variables overlap)


class UvicornWorker(BaseUvicornWorker):
    """
    Configuration for uvicorn workers.

    This class is subclassing UvicornWorker and defines
    some parameters class-wide, because it's impossible,
    to pass these parameters through gunicorn.
    """

    CONFIG_KWARGS = {  # noqa: WPS115 (upper-case constant in a class)
        "loop": "uvloop" if uvloop is not None else "asyncio",
        "http": "httptools",
        "lifespan": "on",
        "factory": True,
        "proxy_headers": False,
    }


class GunicornApplication(BaseApplication):
    """
    Custom gunicorn application.

    This class is used to start guncicorn
    with custom uvicorn workers.
    """

    def __init__(  # noqa: WPS211 (Too many args)
        self,
        app: str,
        host: str,
        port: int,
        workers: int,
        **kwargs: Any,
    ):
        self.options = {
            "bind": f"{host}:{port}",
            "workers": workers,
            "worker_class": "verifire_service.gunicorn_runner.UvicornWorker",
            **kwargs,
        }
        self.app = app
        super().__init__()

    def load_config(self) -> None:
        """
        Load config for web server.

        This function is used to set parameters to gunicorn
        main process. It only sets parameters that
        gunicorn can handle. If you pass unknown
        parameter to it, it crash with error.
        """
        for key, value in self.options.items():
            if key in self.cfg.settings and value is not None:
                self.cfg.set(key.lower(), value)

    def load(self) -> str:
        """
        Load actual application.

        Gunicorn loads application based on this
        function's returns. We return python's path to
        the app's factory.

        :returns: python path to app factory.
        """
        return import_app(self.app)


```

**Comment by nonchan7720** [comment]
> it works as needed , may be becuase the fast-api template i am using already is running an event loop of uvicorn uvloop ? 🤔

I was able to confirm that the error occurs when an event has already been started, so I

PRs were merged, as shown in the sample code in the previous comment.
I think it is possible to poll by calling the `handler` method as in the sample code in the previous comment.

**Comment by nonchan7720** [comment]
I will change the sample code to use `@app.on_event("startup")`.

**Comment by grizzly-monkey** [comment]
ok can you release it to pypi so that i can refresh dep and check...
please excuse my python brevity as i am novice py dev.

Thanks for all your help.

**Comment by nonchan7720** [comment]
I've confirmed that it works fine as shown here, so I'll merge it and release it: 🙆‍♂️
https://github.com/nonchan7720/sqs-polling/pull/24#issuecomment-1793716535

**Comment by nonchan7720** [comment]
Released by v0.0.5.
